File: mahsoul_dz/lib/data/models/shared/notification.dart
class NotificationModel {
  final String id;
  final String farmerId;
  final String orderId;
  final String title;
  final String message;
  final String type;
  final bool isRead;
  final int createdAt;
  final Map<String, dynamic>? order;

  NotificationModel({
    required this.id,
    required this.farmerId,
    required this.orderId,
    required this.title,
    required this.message,
    required this.type,
    required this.isRead,
    required this.createdAt,
    this.order,
  });

  factory NotificationModel.fromJson(Map<String, dynamic> json) {
    return NotificationModel(
      id: json['id'] as String,
      farmerId: json['farmer_id'] as String,
      orderId: json['order_id'] as String,
      title: json['title'] as String,
      message: json['message'] as String,
      type: json['type'] as String,
      isRead: json['is_read'] as bool,
      createdAt: json['created_at'] as int,
      order: json['order'] as Map<String, dynamic>?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'farmer_id': farmerId,
      'order_id': orderId,
      'title': title,
      'message': message,
      'type': type,
      'is_read': isRead,
      'created_at': createdAt,
      'order': order,
    };
  }
}

File: mahsoul_backend/app/models/notification.py
from app.utils.database import db
from datetime import datetime

class Notification(db.Model):
    __tablename__ = 'notifications'
    
    id = db.Column(db.String(50), primary_key=True)
    farmer_id = db.Column(db.String(50), db.ForeignKey('users.id'), nullable=False)
    order_id = db.Column(db.String(50), db.ForeignKey('orders.id'), nullable=False)
    title = db.Column(db.String(200), nullable=False)
    message = db.Column(db.Text, nullable=False)
    type = db.Column(db.String(50), default='order')  # 'order', 'system', etc.
    is_read = db.Column(db.Boolean, default=False)
    created_at = db.Column(db.BigInteger, nullable=False)
    
    # Relationships
    farmer = db.relationship('User', backref='notifications')
    order = db.relationship('Order', backref='notifications')
    
    def to_dict(self):
        return {
            'id': self.id,
            'farmer_id': self.farmer_id,
            'order_id': self.order_id,
            'title': self.title,
            'message': self.message,
            'type': self.type,
            'is_read': self.is_read,
            'created_at': self.created_at,
            'order': {
                'id': self.order.id,
                'status': self.order.status,
                'total_price': self.order.total_price,
            } if self.order else None
        }